### PR TITLE
Update core to be CLI - Closes #257 and #258

### LINKED ISSUE #257: Update core structure become a CLI
### Description

Update the `lisk-core` structure to become a CLI using the [oclif](https://oclif.io) framework.

### Motivation

In order to achieve #254, this preparation is required.

### Acceptance Criteria

- The folder structure should be:

```
/src
  /commands
  /application
   - /transactions
   - /utils
   - index.ts
```

- The tests should also be setup using the [oclif](https://oclif.io) framework

### Additional Information

Related issues: #254

### LINKED ISSUE #258: Add core help, core warranty, core copyright and core version commands
### Description

Add `help`, `warranty`, `copyright` and `version` commands.

### Motivation

Migrate the standard `core` commands from Lisk Commander.

### Acceptance Criteria

- Add commands and minimal tests using oclif (see: https://oclif.io/docs/testing)

### Additional Information

Related issues: #254

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?
This PR resolves #257 and resolves #258 

### How was it solved?
- Added oclif setup
- Add copryright and warranty command
- Setup command tests
- Format src and test

### How was it tested?
- Change package.json lisk-sdk version to 2.3.8
- Remove engine-strict in `.npmrc` temporally
- npm i
- cd sdkDir/sdk; yarn link; cd coreDir; yarn link lisk-sdk 
- npm run lint
- npm run test
- ./bin/run copyright
- ./bin/run warranty

### CODE REVIEW

File: src/application/index.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ */
+import { Application } from 'lisk-sdk';
+
+// Temporally disable eslint
+/* eslint-disable */
+export const getApplication = (genesisBlock: any, config: any): Application => {
+	/**
+	 * We have to keep it in try/catch block as it can throw
+	 * exception while validating the configuration
+	 */
```
Comment (ManuGowda) [inline src/application/index.ts:22]: try catch missing or this comment is obsolete?

File: src/commands/copyright.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/*
+ * LiskHQ/lisk-commander
```
Comment (ManuGowda) [inline src/commands/copyright.ts:2]: lisk-core? if yes, then same in below reference too

File: src/application/index.ts (hunk 1 above)
Comment (shuse2) [reply src/application/index.ts:22]: yes the comment is obsolete, will remove

**Comment by shuse2** [comment]
@pablitovicente 
Oclif uses fancy-test by default, which is helper for test commands, and we if we use that, we cannot use Jest

File: test/commands/copyright.spec.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/*
+ * LiskHQ/lisk-commander
```
Comment (ManuGowda) [inline test/commands/copyright.spec.ts:2]: rename

File: test/commands/warranty.spec.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/*
+ * LiskHQ/lisk-commander
```
Comment (ManuGowda) [inline test/commands/warranty.spec.ts:2]: rename

File: test/setup.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
 /*
+ * LiskHQ/lisk-commander
```
Comment (ManuGowda) [inline test/setup.js:2]: rename

**Pull request merged**